src/upload/upload.ts

```diff
@@ -0,0 +1,50 @@
+import { MongoClient, Collection } from "mongodb";
+
+interface document {}
+type documents = Array<document>;
+interface uploadData {
+  [key: string]: documents;
+}
+interface uploadConfig {
+  dbName: string;
+  mongoURI: string;
+  data: uploadData;
+}
+
+async function uploadToDatabase(params: uploadConfig): Promise<void> {
+  const dbName: string = params.dbName;
+  const mongoURI: string = params.mongoURI;
+  const data: uploadData = params.data;
+  const client: MongoClient = new MongoClient(mongoURI, {
+    useUnifiedTopology: true,
+  });
+
+  try {
+    console.log("connecting....");
+    await client.connect((err: Error, client) => {
+      console.log("connected.");
+      if (err != null) {
+        console.log("Error connecting to database");
+        console.log(err);
+        return;
+      }
+      const db = client.db(dbName);
+      Object.keys(data).forEach((collectionName: string & keyof uploadData) => {
+        if (data[collectionName] == null || data[collectionName].length == 0) {
+          console.log(
+            collectionName + " has no data to be uploaded. Skipping...",
+          );
+          return;
+        }
+
+        const collection: Collection = db.collection(collectionName);
+        collection.insertMany(data[collectionName]);
+        console.log(collectionName + " uploaded successfully");
+      });
+    });
+  } catch (err) {
+    console.log(err);
+  }
```

Comment:
```suggestion
	await client.connect();
    const db = client.db(dbName);	
    Object.keys(data).forEach((collectionName: string & keyof uploadData) => {
      if (data[collectionName] === null || data[collectionName].length === 0) {
        console.log(
          collectionName + " has no data to be uploaded. Skipping...",
        );
        return;
      }

      const collection: Collection = db.collection(collectionName);
      collection.insertMany(data[collectionName]);
      console.log(collectionName + " uploaded successfully");
    });
  } catch (err) {
    console.log(err);
  }
```

With await we don't need to provide a callback

Reply:
Somehow this didn't work for me. Will try again.

Reply:
The code I've suggested probably isn't accurate. I just saw the doc which said, `await client.connect()` is enough to make a connection, which throws an exception if it fails

Reply:
Changed. Take a look.